tests/unit/test_akafka.py

```diff
@@ -16,67 +16,38 @@
 
 """Testing Apache Kafka based providers."""
 
+from collections import namedtuple
 from contextlib import nullcontext
 from typing import Optional
-from unittest.mock import MagicMock, Mock
+from unittest.mock import AsyncMock, Mock
 
 import pytest
 
 from hexkit.custom_types import JsonObject
 from hexkit.providers.akafka import KafkaEventPublisher, KafkaEventSubscriber
-from hexkit.utils import NonAsciiStrError
 
 
-@pytest.mark.parametrize(
-    "type_, key, topic, expected_headers, exception",
-    [
-        ("test_type", "test_key", "test_topic", [("type", b"test_type")], None),
-        (
-            "test_ßtype",  # non ascii
-            "test_key",
-            "test_topic",
-            [("type", b"test_type")],
-            NonAsciiStrError,
-        ),
-        (
-            "test_type",
-            "test_ßkey",  # non ascii
-            "test_topic",
-            [("type", b"test_type")],
-            NonAsciiStrError,
-        ),
-        (
-            "test_type",
-            "test_key",
-            "test_ßtopic",  # non ascii
-            [("type", b"test_type")],
-            NonAsciiStrError,
-        ),
-    ],
-)
-def test_kafka_event_publisher(
-    type_: str,
-    key: str,
-    topic: str,
-    expected_headers: list[tuple[str, bytes]],
-    exception: Optional[type[Exception]],
-):
+@pytest.mark.asyncio
+async def test_kafka_event_publisher():
```

Comment:
This is a good point. I was thinking a bit more and probably it's either way not a good design to rely on the implementer (in case of the publishing protocol that is us, but in case of the subscription protocol that is whomever is using our library) to call super().

Maybe it is cleaner to have the validation logic in a non-abstract method which is calling another private method. This private method is marked as abstract and must be implemented by the developer that subclasses from that abstract class.  
Like this we would nicely separate the concerns of users and implementors in two separate methods.

Please have a look at this commit, as an example: https://github.com/ghga-de/hexkit/pull/12/commits/d599600b66ff7838ed85da4e231cc99af3a9fe0e

Reply:
Yes, that's a nice solution. +1